src/server/debug.rs

```diff
@@ -347,6 +353,66 @@ impl Debugger {
         Ok(res)
     }
 
+    pub fn unsafe_conf_change(&self, conf_change: UnsafeConfChange) -> Result<Vec<(u64, Error)>> {
+        match conf_change {
+            UnsafeConfChange::RemoveStores(store_ids) => self.remove_failed_stores(store_ids),
+        }
+    }
+
+    fn remove_failed_stores(&self, store_ids: Vec<u64>) -> Result<Vec<(u64, Error)>> {
+        let db = &self.engines.kv_engine;
+        let handle = box_try!(get_cf_handle(db.as_ref(), CF_RAFT));
+        let read_opts = IterOption::new(
+            Some(keys::REGION_META_MIN_KEY.to_owned()),
+            Some(keys::REGION_META_MAX_KEY.to_owned()),
+            false,
+        ).build_read_opts();
+        let mut iter = DBIterator::new_cf(Arc::clone(db), handle, read_opts);
+        iter.seek(SeekKey::from(keys::REGION_META_MIN_KEY));
+
+        let wb = WriteBatch::new();
+        let store_ids = HashSet::<u64>::from_iter(store_ids);
+        let mut errors = Vec::with_capacity(store_ids.len());
+
+        for (key, value) in &mut iter {
+            let (region_id, suffix_type) = box_try!(keys::decode_region_meta_key(&key));
+            if suffix_type != keys::REGION_STATE_SUFFIX {
+                continue;
+            }
+            let mut region_state = RegionLocalState::new();
+            if let Err(e) = region_state.merge_from_bytes(&value) {
+                errors.push((region_id, box_err!(e)));
```

Comment:
It looks like corrupted, I think it is better to skip writing the `wb` if we meet any error, so things will not get complicated hopefully.

Reply:
I think we should return the error and skip writing the `wb`.

Reply:
I think `skip write the wb` is good. But maybe collect all errors in one call and return them is better?

Reply:
I think return error here is enough.

Reply:
It's enough but I think collect errors for all regions is better, because if any error happends, we can fix them after run the command 1 time. Otherwise we need to run the command many times to find all errors and fix them.

Reply:
We should not meet such error in general, as well as the error at L384. So just return early is simple and more straightforward.